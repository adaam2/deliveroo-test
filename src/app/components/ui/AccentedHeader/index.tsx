import React from 'react';

import styles from "./AccentedHeader.module.scss";
import { Container } from 'ui';

export interface AccentedHeaderProps {
  children: any;
  className?: string;
}

const AccentedHeader: React.SFC<AccentedHeaderProps> = ({ children, className }) => {
  return (
    <header className={styles.accentedHeader}>
      <Container className={className}>
        {children}
      </Container>
    </header>
  );
}

export default AccentedHeader;