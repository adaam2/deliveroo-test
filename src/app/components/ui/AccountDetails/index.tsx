import React from 'react';

import { Icon, P } from '../';
import styles from "./AccountDetails.module.scss";

export interface AccountDetailsProps {
  userName: string;
}

const AccountDetails: React.SFC<AccountDetailsProps> = ({ userName }) => {
  return (
    <div className={styles.wrapper}>
      <div className={styles.flexItem}>
        <Icon iconKey="user" />
      </div>
      <div className={styles.userName}>
        <P marginLeft={8} fontSize={16} colour="#2e3333">
          {userName}
        </P>
      </div>
    </div>
  );
}

export default AccountDetails;