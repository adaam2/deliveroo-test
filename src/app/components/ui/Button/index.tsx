import React from 'react';
import classNames from "classnames/bind";

import styles from "./Button.module.scss";

const cx = classNames.bind(styles);

export enum ButtonType {
  Primary,
  Secondary
}

export interface ButtonProps {
  type: ButtonType,
  children: any;
  onClick(e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
  ariaLabel?: string;
}

const Button: React.SFC<ButtonProps> = ({
  type = ButtonType.Primary,
  children,
  onClick,
  ariaLabel
}) => {
  const buttonClasses = cx({
    [styles.button]: true,
    [styles.primary]: type === ButtonType.Primary,
    [styles.secondary]: type === ButtonType.Secondary
  });

  return (
    <button aria-label={ariaLabel} onClick={onClick} className={buttonClasses}>
      {children}
    </button>
  );
}

export default Button;