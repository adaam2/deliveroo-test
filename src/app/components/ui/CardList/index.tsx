import React from 'react';

import styles from "./CardList.module.scss";

export interface CardListProps {
  children: any;
}

const CardList: React.SFC<CardListProps> = ({ children }) => {
  return (
    <div className={styles.list}>
      {children}
    </div>
  );
}

export default CardList;