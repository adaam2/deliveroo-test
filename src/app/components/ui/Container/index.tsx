import React from 'react';
import classNames from "classnames/bind";

import styles from "./Container.module.scss";

const cx = classNames.bind(styles);

export interface ContainerProps {
  children: any;
}

const Container: React.SFC<ContainerProps> = ({ children }) => {
  const containerClasses = () => cx({
    [styles.containerBase]: true
  });

  return (
    <div className={containerClasses()}>
      {children}
    </div>
  );
}

export default Container;