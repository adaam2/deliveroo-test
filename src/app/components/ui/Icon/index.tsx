import React from 'react';

export interface IconProps {
  altText?: string;
}

const Icon: React.SFC<IconProps> = ({ altText }) => {
  const icon = require(`assets/svgs/user.svg`);

  return (
    <>
      <img alt={altText} src={icon} />
    </>
  );
}

export default Icon;