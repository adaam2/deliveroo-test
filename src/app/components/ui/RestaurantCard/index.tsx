import React from 'react';
import { Link } from "react-router-dom";

import { ImageCard, Title, P, Spacer } from 'ui';
import Restaurant from 'objects/restaurant';
import { FontWeight } from '../Typography';

const RestaurantCard: React.SFC<Restaurant> = ({ id, name, imageUrl, price, tags }) => {
  const tagList = tags.map(t => t.title).join(" • ");

  const priceLevel = () => Array.from(Array(price)).map(() => '£').join("");

  return (
    <Link to={`/restaurants/${id}`}>
      <ImageCard altText={name} imageUrl={imageUrl}>
        <Spacer height={8} />
        <Title fontWeight={FontWeight.Bold} fontSize={22} colour="#2e33333">
          {name}
        </Title>
        <Spacer height={5} />
        <P fontWeight={FontWeight.Regular} fontSize={14} colour="#828585">
          {tagList} {tagList.length > 0 ? ' • ' : null} {priceLevel()}
        </P>
      </ImageCard>
    </Link>
  );
}

export default RestaurantCard;