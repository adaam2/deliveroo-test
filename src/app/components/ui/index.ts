import Nav from "./Nav";
import Logo from "./Logo";
import Icon from "./Icon";
import { P, Title } from "./Typography";
import ImageCard from "./ImageCard";
import Container from "./Container";
import Spacer from "./Spacer";
import CardList from "./CardList";
import RestaurantCard from "./RestaurantCard";
import RestaurantList from "./RestaurantList";
import AccentedHeader from "./AccentedHeader";
import AccountDetails from "./AccountDetails";
import Button from "./Button";

export { Nav, Logo, Icon, P, Title, ImageCard, Container, Spacer, CardList, RestaurantCard, AccentedHeader, AccountDetails, RestaurantList, Button };
