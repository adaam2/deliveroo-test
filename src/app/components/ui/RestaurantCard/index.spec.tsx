import React from 'react';
import { mount, ReactWrapper } from "enzyme";

import { RestaurantCard, ImageCard, Title, P } from "ui";
import Restaurant from 'objects/restaurant';

let wrapper: ReactWrapper;

const restaurant: Restaurant = {
  name: "Adam's restaurant",
  imageUrl: "http://placehold.it/240",
  id: "1234",
  price: 2,
  tags: [
    {
      title: "Pizza"
    },
    {
      title: "Burgers"
    }
  ]
};

function setup(
  restaurant: Restaurant
) {
  return mount(
    <RestaurantCard {...restaurant} />
  );
}

describe('<Logo />', () => {
  beforeEach(() => {
    wrapper = setup(restaurant);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an Image Card', () => {
    const imageCard = wrapper.find(ImageCard);

    expect(imageCard).not.toBeNull();
  });

  it('renders the restaurant title', () => {
    const imageCard = wrapper.find(ImageCard);
    const title = imageCard.find(Title);

    expect(title.text()).toEqual(restaurant.name);
  });

  it('renders the expected tag list', () => {
    const imageCard = wrapper.find(ImageCard);
    const subText = imageCard.find(P);

    expect(subText.html())
      .toEqual(expect.stringContaining('Pizza • Burgers'));
  });

  it('renders the price level', () => {
    const imageCard = wrapper.find(ImageCard);
    const subText = imageCard.find(P);

    expect(subText.html()).toEqual(expect.stringContaining('££'));
  });
});
