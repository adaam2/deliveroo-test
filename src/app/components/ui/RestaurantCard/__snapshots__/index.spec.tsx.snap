// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> matches the snapshot 1`] = `
<RestaurantCard
  id="1234"
  imageUrl="http://placehold.it/240"
  name="Adam's restaurant"
  price={2}
  tags={
    Array [
      Object {
        "title": "Pizza",
      },
      Object {
        "title": "Burgers",
      },
    ]
  }
>
  <ImageCard
    altText="Adam's restaurant"
    imageUrl="http://placehold.it/240"
  >
    <div>
      <img
        alt="Adam's restaurant"
        src="http://placehold.it/240"
      />
      <div>
        <Spacer
          height={8}
        >
          <div
            style={
              Object {
                "height": 8,
              }
            }
          />
        </Spacer>
        <Component
          colour="#2e33333"
          fontSize={22}
          fontWeight={600}
        >
          <h1
            style={
              Object {
                "color": "#2e33333",
                "fontSize": 22,
                "fontWeight": 600,
                "lineHeight": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "opacity": undefined,
              }
            }
          >
            Adam's restaurant
          </h1>
        </Component>
        <Spacer
          height={5}
        >
          <div
            style={
              Object {
                "height": 5,
              }
            }
          />
        </Spacer>
        <Component
          colour="#828585"
          fontSize={14}
          fontWeight={400}
        >
          <p
            style={
              Object {
                "color": "#828585",
                "fontSize": 14,
                "fontWeight": 400,
                "lineHeight": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "opacity": undefined,
              }
            }
          >
            Pizza • Burgers
             
             • 
             
            ££
          </p>
        </Component>
      </div>
    </div>
  </ImageCard>
</RestaurantCard>
`;
